test/framework/shootmigrationtest.go

```diff
@@ -96,25 +98,31 @@ func (t *ShootMigrationTest) MigrateShoot(ctx context.Context) error {
 
 	t.MigrationTime = metav1.Now()
 	return t.GardenerFramework.MigrateShoot(ctx, &t.Shoot, t.TargetSeed, func(shoot *gardencorev1beta1.Shoot) error {
-		t := gardencorev1beta1.Toleration{}
-		t.Key = SeedTaintTestRun
-		t.Value = pointer.String(GetTestRunID())
-
-		if shoot.Spec.Tolerations == nil {
-			shoot.Spec.Tolerations = make([]gardencorev1beta1.Toleration, 0)
-		} else {
-			for _, t := range shoot.Spec.Tolerations {
-				if t.Key == SeedTaintTestRun {
-					return nil
-				}
-			}
+		shoot.Spec.Tolerations = appendToleration(shoot.Spec.Tolerations, gardencorev1beta1.SeedTaintProtected, nil)
+		if applyTestRunTaint, err := strconv.ParseBool(t.Config.SeedTaintTestRunValue); applyTestRunTaint && err == nil {
+			shoot.Spec.Tolerations = appendToleration(shoot.Spec.Tolerations, SeedTaintTestRun, pointer.String(GetTestRunID()))
 		}
-		shoot.Spec.Tolerations = append(shoot.Spec.Tolerations, t)
-
 		return nil
 	})
 }
 
+func appendToleration(tolerations []gardencorev1beta1.Toleration, key string, value *string) []gardencorev1beta1.Toleration {
+	fmt.Printf("Applying toleration: %s\n", key)
+	toleration := gardencorev1beta1.Toleration{}
+	toleration.Key = key
+	toleration.Value = value
+	if tolerations == nil {
+		tolerations = make([]gardencorev1beta1.Toleration, 0)
+	} else {
+		for _, t := range tolerations {
+			if t.Key == key {
+				return tolerations
```

Comment:
Why not modify the existing toleration?
```suggestion
				t.Value = value
				return tolerations
```